router/src/infer/v3/scheduler.rs

```diff
@@ -374,6 +392,63 @@ fn filter_send_generations(generations: Vec<Generation>, entries: &mut IntMap<u6
     });
 }
 
+/// Check if block allocations need to be extended
+/// If we don't have enough blocks, request will be filtered with an OutOfPages error
+#[instrument(skip_all)]
+async fn filter_update_allocations(entries: &mut IntMap<u64, Entry>) -> bool {
+    let ids: Vec<u64> = entries
+        .iter()
+        .filter_map(|(id, entry)| {
+            entry
+                .block_allocation
+                .as_ref()
+                .map(|block_allocation| {
+                    if entry.current_length > block_allocation.len() as u32 {
+                        // We need to re-allocate
+                        Some(*id)
+                    } else {
+                        None
+                    }
+                })
+                .unwrap_or(None)
```

Comment:
`map` + `unwrap_or(None)` is called [and_then](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then)